lib/blocks.php

```diff
@@ -224,6 +224,13 @@ function gutenberg_deregister_core_block_and_assets( $block_name ) {
 				}
 			}
 		}
+		if ( ! empty( $block_type->editor_script_handles ) ) {
+			foreach ( $block_type->editor_script_handles as $editor_script_handle ) {
+				if ( str_starts_with( $editor_script_handle, 'wp-block-' ) ) {
+					wp_deregister_script( $editor_script_handle );
```

Comment:
To connect the efforts. I was looking in https://github.com/WordPress/wordpress-develop/pull/5118 at ways to allow referencing the same file path by multiple blocks by storing a handle name in the asset file to use it as a way to prevent registering the same file under different names. As of today, WP core would register the same script file under different handles because the name is generated based on the block name.